cern_access/indico_cern_access/plugin.py

```diff
@@ -0,0 +1,237 @@
+        event = obj
+        registrations = get_requested_registrations(event=event)
+        if registrations:
+            state = send_adams_post_request(event, registrations, update=True)[0]
+            if state == CERNAccessRequestState.accepted:
+                update_access_requests(registrations, state)
+
+    def _registration_form_deleted(self, registration_form):
+        """Withdraws CERN access request for deleted registration form and corresponding registrations"""
+        if registration_form.cern_access_request and registration_form.cern_access_request.is_active:
+            registrations = get_requested_registrations(registration_form.event, regform=registration_form)
+            if registrations:
+                deleted = send_adams_delete_request(registrations)
+                if deleted:
+                    withdraw_access_requests(registrations)
+                    notify_access_withdrawn(registrations)
+            registration_form.cern_access_request.request_state = CERNAccessRequestState.withdrawn
+
+    def _event_deleted(self, event, user):
+        """Withdraws CERN access request for registration forms and corresponding registrations of deleted event"""
+        access_requests_forms = get_requested_forms(event)
+        if access_requests_forms:
+            requested_registrations = get_requested_registrations(event=event)
+            if requested_registrations:
+                deleted = send_adams_delete_request(requested_registrations)
+                if deleted:
+                    withdraw_access_requests(requested_registrations)
+                    notify_access_withdrawn(requested_registrations)
+            for form in access_requests_forms:
+                form.cern_access_request.request_state = CERNAccessRequestState.withdrawn
+
+    def _registration_modified(self, registration, change):
+        """If name of registration changed, updates the ADaMS CERN access request"""
+        if registration.cern_access_request and ('first_name' in change or 'last_name' in change):
+            state, _ = send_adams_post_request(registration.event, [registration], update=True)
+            if state == CERNAccessRequestState.accepted:
+                registration.cern_access_request.request_state = state
+
+    def _event_title_changed(self, event, changes, **kwargs):
+        """Updates event name in the ADaMS CERN access request"""
+        if 'title' not in changes:
+            return
+        requested_registrations = get_requested_registrations(event=event)
+        if requested_registrations:
+            state, _ = send_adams_post_request(event, requested_registrations, update=True)
+            if state == CERNAccessRequestState.accepted:
+                update_access_requests(requested_registrations, state)
+
+    def _is_ticketing_handled(self, regform):
+        """Checks if a registration form has requested access to CERN, if so the plugin will handle tickets mailing """
```

Comment:
trailing whitespace in the docstring (which is also excessively long; a docstring line should be <=72 chars, and only exceed that slightly if wrapping e.g. for a single word would make it look worse)